### PR TITLE
update branch & change base on PRs

### LINKED ISSUE #787: Need support for "Updating the pull request branch with latest upstream changes"
Currently there is no way to update the pull request with the latest upstream changes, when using the library. Github has provided a REST endpoint for it - https://developer.github.com/v3/pulls/#update-a-pull-request-branch

It will be good use case to add to the library.

### PR DESCRIPTION
**Pull request by tginiotis-at-work**
# Description 
Adding a method to update the PR branch from the base branch & a method to change the base of the PR.
Closes #787.

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [x] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [x] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn clean compile` locally. This may reformat your code, commit those changes.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [x] Fill in the "Description" above. 
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHPullRequest.java [hunk 1]
```diff
@@ -565,6 +566,40 @@ public void requestTeamReviewers(List<GHTeam> teams) throws IOException {
...
+    @Deprecated
+    public void updateBranch() throws IOException {
+        owner.root.getClient()
+                .sendRequest(
+                        root.createRequest()
+                                .withPreview(LYDIAN)
+                                .method("PUT")
+                                .with("expected_head_sha", head.getSha())
+                                .withUrlPath(getApiRoute() + "/update-branch"),
+                        null);
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPullRequest.java:600]: Use `fetchHttpStatusCode()`.  Maybe need to verify value is `202` and throw if not.  Not sure.

File: src/main/java/org/kohsuke/github/GHPullRequest.java [hunk 2]
```diff
@@ -565,6 +566,40 @@ public void requestTeamReviewers(List<GHTeam> teams) throws IOException {
...
+     *            the name of the new base branch
+     * @throws IOException
+     *             the io exception
+     */
+    public void setBaseBranch(String newBaseBranch) throws IOException {
+        owner.root.getClient()
+                .sendRequest(
+                        root.createRequest().method("PATCH").with("base", newBaseBranch).withUrlPath(getApiRoute()),
+                        null);
+    }
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPullRequest.java:582]: This call returns the updated `GHPullRequest` in JSON. We should parse it and return it.

**Comment by bitwiseman** [comment]
@tginiotis-at-work 
Do you have time to update these?

**Comment by tginiotis-at-work** [comment]
@bitwiseman Hey, yes. I was a bit busy with other stuff, but I will look into these this week.

**Comment by tginiotis-at-work** [comment]
I have added cases for when the base branch does not exist and when the `expected_head_sha` does not match the current head.
Also changed to the test organization.

How does it look now?

**Review by bitwiseman — APPROVED** [review]
Great thanks!

**Pull request merged**